src/DDGeometryCreatorALLEGRO.cc

```diff
@@ -0,0 +1,192 @@
+    this->SetDefaultSubDetectorParameters(*const_cast<dd4hep::rec::LayeredCalorimeterData*>(getExtension( ( dd4hep::DetType::CALORIMETER | dd4hep::DetType::MUON | dd4hep::DetType::ENDCAP), ( dd4hep::DetType::AUXILIARY  |  dd4hep::DetType::FORWARD ) )), "MuonEndCap", pandora::MUON_ENDCAP, muonEndCapParameters);
+
+    subDetectorTypeMap[pandora::ECAL_BARREL] = eCalBarrelParameters;
+    subDetectorTypeMap[pandora::ECAL_ENDCAP] = eCalEndCapParameters;
+    subDetectorTypeMap[pandora::HCAL_BARREL] = hCalBarrelParameters;
+    subDetectorTypeMap[pandora::HCAL_ENDCAP] = hCalEndCapParameters;
+    subDetectorTypeMap[pandora::MUON_BARREL] = muonBarrelParameters;
+    subDetectorTypeMap[pandora::MUON_ENDCAP] = muonEndCapParameters;
+
+    // FIXME! AD: currently ignoring tracker parameters since we are using truth tracks
+    /*
+    PandoraApi::Geometry::SubDetector::Parameters trackerParameters;
+
+    trackerParameters.m_subDetectorName = "Tracker";
+    trackerParameters.m_subDetectorType = pandora::INNER_TRACKER;
+    trackerParameters.m_innerRCoordinate = getTrackingRegionExtent()[0];
+    trackerParameters.m_innerZCoordinate = 0.f;
+    trackerParameters.m_innerPhiCoordinate = 0.f;
+    trackerParameters.m_innerSymmetryOrder = 0;
+    trackerParameters.m_outerRCoordinate = getTrackingRegionExtent()[1];
+    trackerParameters.m_outerZCoordinate = getTrackingRegionExtent()[2];
+    trackerParameters.m_outerPhiCoordinate = 0.f;
+    trackerParameters.m_outerSymmetryOrder = 0;
+    trackerParameters.m_isMirroredInZ = true;
+    trackerParameters.m_nLayers = 0;
+    subDetectorTypeMap[pandora::INNER_TRACKER] = trackerParameters;
+    */
+
+    ///FIXME:Implement a parameter for the Coil/Solenoid name
+    ///NOTE: Is this the way to go, or should we go with reco structure?
+    try{
+        PandoraApi::Geometry::SubDetector::Parameters coilParameters;
+
+        /*
+        const dd4hep::rec::LayeredCalorimeterData * coilExtension= getExtension( ( dd4hep::DetType::COIL ) );
+        coilParameters.m_subDetectorName = "Coil";
+        coilParameters.m_subDetectorType = pandora::COIL;
+        coilParameters.m_innerRCoordinate = coilExtension->extent[0]/ dd4hep::mm; 
+        coilParameters.m_innerZCoordinate = 0.f;
+        coilParameters.m_innerPhiCoordinate = 0.f;
+        coilParameters.m_innerSymmetryOrder = 0;
+        coilParameters.m_outerRCoordinate = coilExtension->extent[1]/ dd4hep::mm;
+        coilParameters.m_outerZCoordinate = coilExtension->extent[3]/ dd4hep::mm;
+        coilParameters.m_outerPhiCoordinate = 0.f;
+        coilParameters.m_outerSymmetryOrder = 0;
+        coilParameters.m_isMirroredInZ = true;
+        coilParameters.m_nLayers = 0;
+        subDetectorTypeMap[pandora::COIL] = coilParameters;
+        */
+
+        // FIXME! AD: below an ugly way is used to define the geometry of the Solenoid since we do not have the LayeredCalorimeterData for Coil in ALLEGRO
```

Comment:
Why not add this to the coil?

Reply:
I am not sure if I understand your question. 
Do you mean why we do not have CaloExtension for COIL?

Reply:
Yes.

Reply:
Yes, there is currently no actual coil, it is accounted for in the material of the outer radius side of the cryostat